### PR TITLE
Support routing

### PR DESCRIPTION
**Pull request by itsJess1ca**
Adjusted publish to use amqp's publish method vs sendToQueue, allowing for much more control over routing.

Also added `bindQueue` and `unbindQueue` for assigning queues to exchanges.

### CODE REVIEW

**Comment by itsJess1ca** [comment]
Looks like CI failed due to `chai` not being in dependencies :/ 
If that's a blocker for merge, let me know steps you'd like taken to get this merged and I'll do what I can

**Comment by shaunburdick** [comment]
@j3ddesign Thank your for your contribution!
The CI failed due to a new lint rule. I'll get that fixed in a little [bit](https://github.com/swimlane/lint-rules/pull/1).

File: src/queue.ts [hunk 1]
```diff
@@ -82,13 +83,13 @@ export class AmqpQueue<T> extends EventEmitter {
    * @returns {Promise<{ content: any; properties: PublishOptions }>}
    * @memberof AmqpQueue
    */
+  // tslint:disable-next-line:max-line-length
```
Comment (shaunburdick) [inline src/queue.ts]: I don't think this is needed, this line is just shy of the 120 mark

File: src/queue.ts [hunk 2]
```diff
@@ -97,7 +98,16 @@ export class AmqpQueue<T> extends EventEmitter {
       content = new Buffer(json);
     }
 
-    chnl.sendToQueue(this.options.name, content, opts);
+    const exchange = options.exchangeOverride === '' || options.exchangeOverride ?
```
Comment (shaunburdick) [inline src/queue.ts:100]: Wouldn't this always apply `options.exchangeOverride`? It defaults as `''` so this would always be true.

**Review by shaunburdick — CHANGES_REQUESTED** [review]

File: src/queue.ts (hunk 2 above)
Comment (itsJess1ca) [reply src/queue.ts:100]: `options.exchangeOverride` is the options passed into the `publish` method, which defaults as `{}`
The exchange defaulted to `''` is `this.options.exchange` ;)

The goal here is to allow `exchange` to be set to whatever is primarily used, and then on particular publish calls, let the consumer override the exchange.
For example, on a personal project I send to a `sockets` exchange, but sometimes need to publish to `''`

File: src/queue.ts (hunk 1 above)
Comment (itsJess1ca) [reply src/queue.ts]: Woops, missed that one. The line was longer when I did first pass. Removed it now 👍

File: src/queue.ts (hunk 2 above)
Comment (shaunburdick) [reply src/queue.ts:100]: Fair enough, got the `exchange` on line 12 mixed up with `exchangeOverride` (yay work travel)
LGTM! Thanks again!

**Review by shaunburdick — APPROVED** [review]

**Pull request merged**